Fix 'i' character being displayed

beauwilliams opened the pull request:
When we enter insert mode in neovim the 'i' character appears at the start of every search due to the feedcharacter function being used currently. For example when we select to search through 'Vim Commands' we need to backspace the 'i' character away each time before we can start typing in some text -- unless our search string starts with i of course :)
We can probably remove the entire neovim logic here and make the code more portable but that will need just a few more tweaks
Perhaps earlier versions of nvim did not support `startinsert` command and is the reason for this added logic, however I have tested on the nightly 5.0 version release as of December 20 and it is now working correctly. These changes will fix the problem as is and should be okay for now.

beauwilliams committed:
Fix 'i' character being displayed

When we enter insert mode in neovim the 'i' character appears at the start of every search due to the feedcharacter function being used currently. For example when we select to search through 'Vim Commands' we need to backspace the 'i' character away each time before we can start typing in some text -- unless our search string starts with i of course :)
We can probably remove the entire neovim logic here and make the code more portable but that will need just a few more tweaks
Perhaps earlier versions of nvim did not support `startinsert` command and is the reason for this added logic, however I have tested on the nightly 5.0 version release as of December 20 and it is now working correctly. These changes will fix the problem as is and should be okay for now.

laher commented:
Thanks for this, I'm glad that nvim has fixed this. 

It seems like the fix is in nvim 0.5+ only. I don't want to break fuzzymenu for nvim 0.4.x - I'm on 0.4.4, via latest Ubuntu. Ubuntu LTS installs 0.4.3 I think.

I think this change needs a `has('nvim-0.5')` around it. Alternatively, something like `"startinsert" -> "if mode is not insert mode" -> "feedkeys(i)"` might be a more sensible test . Not sure.

I'll test it out later this evening or this week. I'm not too experienced with managing different versions of nvim, but it's about time I tried.

beauwilliams commented:
Great suggestion, that's a much better way to do it.. I can give version checking for 0.5 a shot and if it works we probably can push it out without having to wait for 0.5 release then :)

Will let you know how I go

beauwilliams committed:
Added check for nvim 0.5 

This logic should pick up the 0.5 nightly version of nvim that supports `startinsert` command. Else it will revert for feedkeys for all other neovim versions below 0.5. We may need to revise in the future versions of nvim. But at this point that looks some time away.

Tested on 0.5 and works for me. Will need testing on other versions of nvim/vim to confirm all is well.

beauwilliams commented:
I've tested and pushed a revised version as discussed. This is just a basic fix. It will check for 0.5 and otherwise will revert to `feedkeys` for all other nvim versions

laher commented:
thanks, it works fine for me on 0.4.x :clap:

laher commented:
btw, in case it wasn't clear, `has('nvim-0.5')` will resolve to true for future versions like 0.6, 0.7, ...

From my nvim, `:echo has('nvim-0.4')` and  `:echo has('nvim-0.3')` both return `1`

beauwilliams commented:
Aha perfect.. That makes this issue a lot easier then for us. Looks like we might be okay to roll then with what we have now :) 

Great to hear testing went good on your end!

laher commented:
yup, all merged now thanks :shipit: :rocket:
